include/eve/detail/function/simd/common/load.hpp

```diff
@@ -11,11 +11,95 @@
 #include <eve/detail/abi.hpp>
 #include <eve/detail/function/bit_cast.hpp>
 #include <eve/memory/aligned_ptr.hpp>
+#include <eve/memory/pointer.hpp>
+#include <eve/function/replace.hpp>
+#include <eve/function/safe.hpp>
+#include <eve/function/unsafe.hpp>
 
 #include <iterator>
 
 namespace eve::detail
 {
+  //================================================================================================
+  // Take a data_source and a Cardinal to determine which wide to build
+  //================================================================================================
+  template<data_source Ptr, typename Cardinal = void>
+  struct loaded_wide : as_wide<typename pointer_traits<Ptr>::value_type, Cardinal>
+  {};
+
+  template<data_source Ptr>
+  struct loaded_wide<Ptr> : as_wide<typename pointer_traits<Ptr>::value_type>
+  {};
+
+  template<data_source Ptr, typename Cardinal = void>
+  using loaded_wide_t = typename loaded_wide<Ptr,Cardinal>::type;
```

Comment:
I'd prefer to call it `wide_value_type_t` tbh

Reply:
Fine